### PR TITLE
emby-server(-devel): Synchronize Emby Makefiles, Automate plist and Add Missing Libraries

### PR DESCRIPTION
**Pull request by ConstrictM**
- emby-sever and emby-server-devel are nearly identical, so using emby-server as the basis for -devel
- Automate plist
- Address poudriere staging library dependency warnings
[emby-server-4.7.11.0_4.log](https://github.com/freebsd/freebsd-ports/files/10797594/emby-server-4.7.11.0_4.log)
[emby-server-devel-4.8.0.21_4.log](https://github.com/freebsd/freebsd-ports/files/10797758/emby-server-devel-4.8.0.21_4.log)

### CODE REVIEW

**Comment by jbeich** [comment]
Requires review/approval by @driesmp (maintainer). Looks like this hasn't been submitted via bugzilla, so cannot land under "maintainer timeout".

File: multimedia/emby-server/Makefile [hunk 1]
```diff
@@ -50,12 +50,24 @@ LIB_DEPENDS=	libaribb24.so:multimedia/aribb24 \
...
+		libtiff.so:graphics/tiff \
+		liblcms2.so:graphics/lcms2 \
+		libOpenEXR-3_1.so:graphics/openexr \
+		libglib-2.0.so:devel/glib20 \
+		libgio-2.0.so:devel/glib20 \
+		libgobject-2.0.so:devel/glib20 \
+		libpangocairo-1.0.so:x11-toolkits/pango \
+		libpango-1.0.so:x11-toolkits/pango \
+		libpangoft2-1.0.so:x11-toolkits/pango \
+		libcairo.so:graphics/cairo
```
Comment (driesmp) [inline multimedia/emby-server/Makefile:64]: Lets sort these libs alphabetically

File: multimedia/emby-server/Makefile [hunk 2]
```diff
@@ -50,12 +50,24 @@ LIB_DEPENDS=	libaribb24.so:multimedia/aribb24 \
...
 RUN_DEPENDS=	icu>0:devel/icu \
 		libinotify>0:devel/libinotify \
 		libunwind>0:devel/libunwind
 
-USES=		compiler:c11 gmake iconv localbase:ldflags pkgconfig sqlite ssl:run
+USES=		compiler:c11 gmake iconv localbase:ldflags pkgconfig sqlite ssl:run jpeg gettext-runtime
```
Comment (driesmp) [inline multimedia/emby-server/Makefile:69]: Same for the uses, sort alphabetically

File: multimedia/emby-server/Makefile [hunk 3]
```diff
@@ -50,12 +50,24 @@ LIB_DEPENDS=	libaribb24.so:multimedia/aribb24 \
...
+USE_GNOME=	glib20 pango cairo
```
Comment (driesmp) [inline multimedia/emby-server/Makefile:70]: Same for USE_GNOME, sort alphabetically

**Comment by driesmp** [comment]
> Requires review/approval by @driesmp (maintainer). Looks like this hasn't been submitted via bugzilla, so cannot land under "maintainer timeout".

Thanks for the ping! Anoying that GH doesnt send a PM to maintainer yet :-)

File: multimedia/emby-server/Makefile (hunk 1 above)
Comment (ConstrictM) [reply multimedia/emby-server/Makefile]: Done

File: multimedia/emby-server/Makefile (hunk 2 above)
Comment (ConstrictM) [reply multimedia/emby-server/Makefile]: Done

File: multimedia/emby-server/Makefile (hunk 3 above)
Comment (ConstrictM) [reply multimedia/emby-server/Makefile]: Done

**Comment by driesmp** [comment]
Actually, could we split them up like this:

LIB_DEPENDS= libraries that are directly used by the configure target
LIB_DEPENDS+= the other libraries poudriere complains about
USES= USES that are directly used by the configure target
USES+= all the other uses poudriere complains about


The configure target is this one;: 		

--enable-chromaprint \
		--enable-fontconfig \
		--enable-gnutls \
		--enable-gpl \
		--enable-iconv \
		--enable-libaribb24 \
		--enable-libass \
		--enable-libdav1d \
		--enable-libdrm \
		--enable-libfreetype \
		--enable-libfribidi \
		--enable-libmfx \
		--enable-libopus \
		--enable-libtesseract \
		--enable-libtheora \
		--enable-libvorbis \
		--enable-libvpx \
		--enable-libwebp \
		--enable-libx264 \
		--enable-libx265 \
		--enable-libzvbi \
		--enable-opencl \
		--enable-optimizations \
		--enable-static \
		--enable-vaapi \
		--enable-version3

**Comment by driesmp** [comment]
Seems like we are missing some extra libs from the pre-shipped libraries, should also add them.

[/usr/ports/multimedia/emby-server/work/lib] ldd ./libvips.so |grep "not found"
        libimagequant.so.0 => not found (0)
        libexif.so.12 => not found (0)
        libopenjp2.so.7 => not found (0)
        liborc-0.4.so.0 => not found (0)
        libheif.so.1 => not found (0)
        libjxl.so.0.7 => not found (0)
        libjxl_threads.so.0.7 => not found (0)

**Comment by ConstrictM** [comment]
> Seems like we are missing some extra libs from the pre-shipped libraries, should also add them, libSkiaSharp seems OK.
> 
> [/usr/ports/multimedia/emby-server/work/lib] ldd ./libvips.so |grep "not found" libimagequant.so.0 => not found (0) libexif.so.12 => not found (0) libopenjp2.so.7 => not found (0) liborc-0.4.so.0 => not found (0) libheif.so.1 => not found (0) libjxl.so.0.7 => not found (0) libjxl_threads.so.0.7 => not found (0)

@driesmp Done.

**Comment by ConstrictM** [comment]
Screen Shot 2023-02-28 at 8 34 40 PM
Is this an issue? I'm not sure how to get the 0.7 version

**Comment by ConstrictM** [comment]
Hmmm getting an error in emby with this package

```
2023-02-28 20:50:22.207 Info Server: http/1.1 POST http://‌‍‍10.0.0.35‌:8096/emby/ScheduledTasks/Running/41025d42ea034a89558382b16c965757?X-Emby-Client=Emby Web&X-Emby-Device-Name=Chrome macOS&X-Emby-Device-Id=44cc8505-f996-44f4-a253-aaae5ea42a06&X-Emby-Client-Version=4.8.0.21&X-Emby-Language=en-us. UserAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36 Edg/110.0.1587.57
2023-02-28 20:50:22.208 Info Server: http/1.1 Response 204 to ‌‍‍10.0.0.109‌. Time: 2ms. http://‌‍‍10.0.0.35‌:8096/emby/ScheduledTasks/Running/41025d42ea034a89558382b16c965757?X-Emby-Client=Emby Web&X-Emby-Device-Name=Chrome macOS&X-Emby-Device-Id=44cc8505-f996-44f4-a253-aaae5ea42a06&X-Emby-Client-Version=4.8.0.21&X-Emby-Language=en-us
2023-02-28 20:50:22.222 Error TaskManager: Error
	*** Error Report ***
	Version: 4.8.0.21
	Command line: /usr/local/lib/emby-server/system/EmbyServer.dll -os freebsd -ffdetect /usr/local/lib/emby-server/bin/ffdetect -ffmpeg /usr/local/lib/emby-server/bin/ffmpeg -ffprobe /usr/local/lib/emby-server/bin/ffprobe -programdata /config
	Operating system: FreeBSD 13.1-RELEASE-p2 FreeBSD 13.1-RELEASE-p2 n245412-484f039b1d0 TRUENAS
	Framework: .NET 6.0.5
	OS/Process: x64/x64
	Runtime: usr/local/lib/emby-server/system/System.Private.CoreLib.dll
	Processor count: 12
	Data path: /config
	Application path: /usr/local/lib/emby-server/system
	SQLitePCL.pretty.SQLiteException: Error: no such column:  - select A.Id from MediaItems A join AncestorIds2 on A.Id=AncestorIds2.ItemId AND AncestorId in (329579) where A.Type in (8) AND RunTimeTicks>=@MinRunTimeTicks AND ((Select Id from MediaItems where AlbumId=A.AlbumId and PresentationUniqueKey <> A.PresentationUniqueKey limit 1) not NULL) AND ((select StartPositionTicks from Chapters3 where Chapters3.ItemId=A.Id and MarkerType in (1) limit 1) is null) AND Coalesce((select Value from ItemExtraData where ItemExtraData.ItemId=A.Id and ExtradataTypeId=2 limit 1), "") <> Cast(A.DateModified as Text) AND IsVirtualItem=@IsVirtualItem
	SQLitePCL.pretty.SQLiteException: Exception of type 'SQLitePCL.pretty.SQLiteException' was thrown.
	   at SQLitePCL.pretty.SQLiteDatabaseConnection.PrepareStatement(String sql, String& tail)
	   at SQLitePCL.pretty.DatabaseConnection.PrepareStatement(IDatabaseConnection This, String sql)
	   at Emby.Server.Implementations.Data.SqliteItemRepository.QueryItemsInTransaction[T](InternalItemsQuery query, IDatabaseConnection db, String[] columnsToSelect, Func`5 rowReaderFn, Boolean singleResult, String methodName)
	   at Emby.Server.Implementations.Data.SqliteItemRepository.GetInternalItemIdsList(InternalItemsQuery query)
	   at Emby.Server.Implementations.Library.LibraryManager.GetInternalItemIds(InternalItemsQuery query)
	   at MediaBrowser.Controller.Entities.Folder.QueryRecursiveForItemIds(InternalItemsQuery query)
	   at MediaBrowser.Controller.Entities.Folder.GetItemIdsInternal(InternalItemsQuery query)
	   at Emby.Providers.Markers.MarkerScheduledTask.Execute(CancellationToken cancellationToken, IProgress`1 progress)
	   at Emby.Server.Implementations.ScheduledTasks.ScheduledTaskWorker.ExecuteInternal(TaskOptions options)
	Source: SQLitePCL.pretty
	TargetSite: SQLitePCL.pretty.IStatement PrepareStatement(System.String, System.String ByRef)
```

**Comment by ConstrictM** [comment]
> Hmmm getting an error in emby with this package when I try to run the tasks for extracting images and detecting intros.

its it possible that this change to [SQLite3 ](https://www.freshports.org/databases/sqlite3/) could have caused this?
_databases/sqlite3: update to 3.41.0_

_Disable option DQS by default._
_Add a note to UPDATING about this._

**Comment by driesmp** [comment]
You can't, add a note to pkg-message for emby-server for now, I'll address this upstream with Emby devs, we need to manually compile sqlite with the option enabled that was turned off by default. Something like:

"For emby-server to work: sqlite has to be built with the DQS option enabled"

**Comment by driesmp** [comment]
You don't have to address this (sqlite error) in this PR, its a completely separate issue.

Is this an issue? I'm not sure how to get the 0.7 version => its fine, this means the pre-shipped libs haven't been recompiled with the more recent version that's available in our ports tree, I will check with upstream.

**Comment by ConstrictM** [comment]
> You don't have to address this (sqlite error) in this PR, its a completely separate issue. There are some mailings about this on the list around the sqlite change. https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=269889
> 
> Is this an issue? I'm not sure how to get the 0.7 version => This means the pre-shipped libs haven't been recompiled with the more recent version that's available in our ports tree, I will check with upstream.

I think that this complete then. Are there any other changes I need to make?

**Comment by driesmp** [comment]
After thinking a bit more about this, one last change I would like to see is that for all libraries that are not required during built (so the libs required by the shipped binaries), we add them as RUN_DEPENDS, you can add them to the existing list, not LIB_DEPENDS. This way the build environment stays clean of unneeded libs. The USES mostly have a ":run" specification to use, see eg: https://cgit.freebsd.org/ports/tree/Mk/Uses/jpeg.mk

**Comment by ConstrictM** [comment]
> After thinking a bit more about this, one last change I would like to see is that for all libraries that are not required during built (so the libs required by the shipped binaries), we add them as RUN_DEPENDS, you can add them to the existing list, not LIB_DEPENDS. This way the build environment stays clean of unneeded libs. The USES mostly have a ":run" specification to use, see eg: https://cgit.freebsd.org/ports/tree/Mk/Uses/jpeg.mk
> 
> Could you rebase against main too?

Done...I think

**Comment by driesmp** [comment]
I just pushed the dedup Makefile bits. I'll take a look at the libs probably sunday.

**Comment by driesmp** [comment]
Ok I pushed both changes. After testing and taking a deeper dive, the poudriere log is a false positive, all needed libs are actually installed as dependencies of the declared dependencies. There were missing some dependencies of the shipped library vips, I added those!

**Pull request closed without merging**